### ISSUE TITLE
[Feature Request] Include `for-syntax` in Syntax Utilities documentation header

### ISSUE THREAD

**Issue by shardulc** [racket/racket#5045]
As a very novice Racket user, I've always interpreted the `require` statement in the header of documentation pages as the statement I should include in my file / evaluate in my REPL to use anything described on that page. But this is not the case for the page about [Syntax Utilities](https://docs.racket-lang.org/reference/syntax-util.html) because (as I discovered after a few minutes of confusion) you need `for-syntax`.

Would it make sense to change the header or include a sentence of clarification to that effect? If so, I can try to open a PR. Open to all suggestions for what the message should exactly say.

Related but distinct issues are #3912 for improving the error message, not the documentation, and #3014, about some confusion about which functions are and should be provided/required `for-syntax`. So maybe a documentation change would help those too.

**Comment by sorawee** [comment]
I don't think it makes sense to change the header, since `(require racket/syntax)` can indeed be used legitimately (e.g. if you are building a library to be required for-syntax, then in the library you need to require racket/syntax relatively in the same phase).

I think adding a sentence to indicate that these utilities are commonly used in a macro, and in such case, require `for-syntax`, could be a good idea. We probably should also add that examples throughout this section assume that `racket/syntax` is `require`d in both phase 0 and 1.

**Comment by mfelleisen** [comment]
We might duplicate the require call in the header libraries commonly used at syntax: 

	`(require racket/syntax)` or `(require (for-syntax racket/syntax))`

It would reduce an obstacle for beginners.

**Comment by rfindler** [comment]
Another idea is a margin note that suggests a `for-syntax` require here (and in `syntax/parse`?) and points to some larger description of phases and what's going on?
